### DISCUSSION TITLE
Is it possible to capture key-up and mouse events from the terminal?

### DISCUSSION THREAD

**Discussion by Oldes** [libuv/libuv#4508] [Q&A]
I'm considering replacing my code with libuv. Is it possible to capture key-up and mouse events from the terminal (cmd)?

**Comment by Oldes** [comment]
On Posix I'm able to enable mouse events using:
```C
write(STDIN_FILENO, "\x1b[?1000h", 10);
```
Then I can see mouse related escape sequences in `tty` events. I wonder if it could work on Windows as well? But it looks there is no mouse related code in the windows related source code:-/

**Reply by bnoordhuis** [reply]
I don't think Windows has anything like that; mouse events are basically window events and those are exposed through the win32 event loop (WndProc and friends.)

**Reply by vtjnash** [reply]
Inside the terminal they arrive as input events. I don't fully know if those get processed by libuv, but the modern terminal likely works the same as unix with regards to enabling it via that write call. Not really a libuv question, since this is terminal emulator specific
